### ISSUE TITLE
ambient enums error with string constants is throwing error

### ISSUE THREAD

**Issue by omouse** [Microsoft/TypeScript#18189]
This angular module, ngx-datatable, cannot be compiled with the latest version of TypeScript (though some people have had success with slightly earlier versions of TypeScript apparently?)

The issue was reported in their github issues: https://github.com/swimlane/ngx-datatable/issues/927#issuecomment-326165165

The enum is declared and each key has a string constant set as its value.

**TypeScript Version:**  2.4.2

**Code**

```ts
export declare enum ColumnMode {
    standard = "standard",
    flex = "flex",
    force = "force",
}
```

**Expected behavior:**

To properly compile.

**Actual behavior:**

TypeScript emits this error message:

```
ERROR in E:/works/ferrari-update/ferrari/Grand/node_modules/@swimlane/ngx-datatable/release/types/column-mode.type.d.ts (2,16): In ambient enum declarations member initializer must be constant expression.
```

That same error appears for all enums. Aren't string constants like `"standard"` constant expressions?

**Comment by kitsonk** [comment]
```
export declare const enum ColumnMode {
    standard = "standard",
    flex = "flex",
    force = "force",
}
```

The problem is non-const enums need to emit actual code, as their values are mutable.  Ambient declarations, by their nature, don't actually emit any run-time code.  This means that they technically need to be `const` enums so all the values of the enums are known to the compiler.

**Comment by mhegazy** [comment]
@omouse i am unable to reproduce this issue. can you please share more context.

**Comment by omouse** [comment]
@kitsonk is there a code example somewhere in the docs you can point me to or could you craft a quick example? I'm a bit confused on this.

@mhegazy when I get home I'll give it another go with typescript 2.4.x and with 2.5.x and let you know.

**Comment by omouse** [comment]
I tried to compile this with versions 2.4.2 and 2.5.x and it worked, I have a feeling there's something else going on with that library or with angular-cli (which uses webpack).

I'll close this for now and re-open if I find more info.

Thanks for the help!
